packages/app/obojobo-document-engine/src/scripts/viewer/util/recommendation-util.js

```diff
@@ -0,0 +1,99 @@
+import NavStore from '../stores/nav-store'
+
+const RecommendationUtil = {
+	boundConfidence(state, page) {
+		if (state.confidenceLevels[page] > 100)
+			state.confidenceLevels[page] = 100
+		if (state.confidenceLevels[page] < 0)
+			state.confidenceLevels[page] = 0
+	},
+
+	onOpenQuestion(state) {
+		if (this.isOnAssessmentPage(state))
+			return
+		state.confidenceLevels[state.currentPageID] += 2
+		this.boundConfidence(state, state.currentPageID)
+	},
+
+	onQuestionAnswered(state, payload) {
+		if (this.isOnAssessmentPage(state))
+			return
+		state.confidenceLevels[state.currentPageID] += payload.value.score === 100 ? 15 : -12.5
```

Comment:
For any 'magic numbers' like 15, -12.5, etc, those should exist as constants (like `const EXAMPLE_CONSTANT = 15`). If they're used in just one file they can sit somewhere at the top - if they're used in more than one place then they should sit in their own constants file which any other file can import.